test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/DacFx/DacFxServiceTests.cs

```diff
@@ -926,5 +930,102 @@ public void VerifyExpectedAndActualBooleanOptionsDictionary(Dictionary<string, D
+
+            VerifyAndCleanup(sqlTable1DefinitionPath);
+            VerifyAndCleanup(sqlTable2DefinitionPath);
+            Directory.Delete(TSqlModelTestFolder);
+
+            Assert.That(model.Version.ToString(), Is.EqualTo(generateTSqlScriptParams.ModelTargetVersion), $"Model version is not equal to {generateTSqlScriptParams.ModelTargetVersion}");
+            Assert.That(objects, Is.Not.Empty);
+
+            var tableNames = objects.Select(o => o.Name.ToString()).ToList();
+
+            Assert.That(tableNames.Count, Is.EqualTo(2), "Model was not populated correctly");
+            CollectionAssert.AreEquivalent(tableNames, new[] { "[dbo].[table1]", "[dbo].[table2]" }, "Table names do not match");
+        }
+
+        /// <summary>
+        /// Verify the generate Tsql model operation, creates an empty model when files are empty
+        /// </summary>
+        [Test]
+        public void GenerateEmptyTSqlModel()
+        {
+            DacFxService service = new DacFxService();
+            Directory.CreateDirectory(TSqlModelTestFolder);
+
+            var generateTSqlScriptParams = new GenerateTSqlModelParams
+            {
+                ProjectUri = Path.Join(TSqlModelTestFolder, "test.sqlproj"),
+                ModelTargetVersion = "Sql160",
+                FilePaths = new string[] { }
+            };
+
+            GenerateTSqlModelOperation op = new GenerateTSqlModelOperation(generateTSqlScriptParams);
+            var model = op.GenerateTSqlModel();
+
+            // clean up
+            Directory.Delete(TSqlModelTestFolder);
+
+            Assert.That(model.GetObjects(DacQueryScopes.UserDefined, ModelSchema.Table).ToList().Count, Is.EqualTo(0), "Model is not empty");
+            Assert.That(model.Version.ToString(), Is.EqualTo(generateTSqlScriptParams.ModelTargetVersion), $"Model version is not equal to {generateTSqlScriptParams.ModelTargetVersion}");
+        }
+
+        /// <summary>
+        /// Verify the generate TSql Model handle
+        /// </summary>
+        [Test]
+        public async Task VerifyGenerateTSqlModelHandle()
+        {
+            DacFxService service = new DacFxService();
+            Directory.CreateDirectory(TSqlModelTestFolder);
+
+            var generateTSqlScriptParams = new GenerateTSqlModelParams
+            {
+                ProjectUri = Path.Join(TSqlModelTestFolder, "test.sqlproj"),
+                ModelTargetVersion = "Sql160",
+                FilePaths = new string[] { }
+            };
+
+            var requestContext = new Mock<RequestContext<ResultStatus>>();
+            requestContext.Setup((RequestContext<ResultStatus> x) => x.SendResult(It.Is<ResultStatus>((result) => result.Success == true))).Returns(Task.FromResult(new object()));
+            await service.HandleGenerateTSqlModelRequest(generateTSqlScriptParams, requestContext.Object);
+
+            Directory.Delete(TSqlModelTestFolder);
```

Comment:
Cleanup stuff should go in a finally so it happens even if the test fails

(or even consider using the TearDown attribute to create a helper that will clean everything up when the tests finish running)